### PR TITLE
add verification to Deploy model

### PR DESCRIPTION
**Pull request by michaelb990**
We need a way to show whether or not a deploy has been verified. This adds a table `verifications` with a foreign key relationship to the deployment they reference. Once a deployment has been verified with `decima-watcher`, the result will be added (append only) as a new verification event.

A couple open questions:
* What is the best way to define verification statuses? This could be a database level constraint, in the code, or in a config file somewhere.
* Is there anything else that would be useful to have in the verifications table?
* Should we add anything else to Decima while we're messing with the DB?

### CODE REVIEW

**Comment by sokrahta** [comment]
there are style issues, run `sbt styleCheck` locally to discover them before submitting to travis-ci

File: src/main/resources/com/socrata/decima/migrate.xml [hunk 1]
```diff
@@ -17,5 +17,11 @@
             <column name="docker_tag" type="varchar(254)"/>
         </addColumn>
     </changeSet>
+    <changeSet author="Michael Brown" id="20150629-add-verification-table">
+        <sqlFile path="com/socrata/decima/migration/20150629-add-verification-table.sql" splitStatements="false" />
```
Comment (sokrahta) [inline src/main/resources/com/socrata/decima/migrate.xml]: liquibase can create a table without separate sql file, e.g.
```xml
   <changeSet author="Michael Brown" id="20150629-add-verification-table">
       <preConditions onFail="MARK_RAN">
           <not>
               <tableExists tableName="verifications"/>
           </not>
       </preConditions>
       <createTable tableName="verifications">
            <column name="id" type="BIGSERIAL">
               <constraints nullable="false" primaryKey="true" unique="true"/>
           </column>
           <column name="status" type="VARCHAR(254)">
               <constraints nullable="false"/>
           </column>
           <column name="details" type="TEXT"/>
           <column name="time" type="TIMESTAMP">
               <constraints nullable="false"/>
           </column>
           <column name="deploy_id" type="BIGINT">
                <constraints references="deploys(id)"/>
           </column>
       </createTable>
   </changeSet>
```

Comment (michaelb990) [inline src/main/resources/com/socrata/decima/migrate.xml]: You're a wizard, Krah :+1:

File: src/main/scala/com/socrata/decima/database/tables/VerificationTable.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package com.socrata.decima.database.tables
+
+import java.sql.Timestamp
+
+import com.socrata.decima.database.DatabaseDriver
+
+trait VerificationTable extends DeployTable {
+  self: DatabaseDriver =>
```
Comment (sokrahta) [inline src/main/scala/com/socrata/decima/database/tables/VerificationTable.scala]: add explicit unit for readability, e.g. `DatabaseDriver => ()`

**Comment by sokrahta** [comment]
If I grok, the intent here is to `curl /version` and compare it with the deployment. 
Will it run against each instance or the app as a whole?

Should probably include a timestamp verifiedAt in the deploy table.
I think appending verified=true on success, and verified=false version={version json from service} on failure, and include the timestamp here.

**Comment by michaelb990** [comment]
@sokrahta this will run against each instance. The verification event will be for the entire service, so my thinking right now is that a partial match will result in a verification status of `SERVICE_MISMATCH` and the `details` field will explain which ones match.

**Pull request merged**